src/main/clm_driver.F90

```diff
@@ -704,16 +705,23 @@ subroutine clm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate, ro
        ! bugs.
        allocate(downreg_patch(bounds_clump%begp:bounds_clump%endp))
        allocate(leafn_patch(bounds_clump%begp:bounds_clump%endp))
-       downreg_patch = bgc_vegetation_inst%get_downreg_patch(bounds_clump)
-       leafn_patch = bgc_vegetation_inst%get_leafn_patch(bounds_clump)
-
        allocate(froot_carbon(bounds_clump%begp:bounds_clump%endp))
        allocate(croot_carbon(bounds_clump%begp:bounds_clump%endp))
-       froot_carbon = bgc_vegetation_inst%get_froot_carbon_patch( &
-            bounds_clump, canopystate_inst%tlai_patch(bounds_clump%begp:bounds_clump%endp))
-       croot_carbon = bgc_vegetation_inst%get_croot_carbon_patch( &
-            bounds_clump, canopystate_inst%tlai_patch(bounds_clump%begp:bounds_clump%endp))
 
+       if(use_fates)then
```

Comment:
I'd add a comment to this about why this is being done. Also try using nan rather than spval.

Reply:
added comments

Reply:
The way to get nan within a scope is to add this use statement:

``` fortran
use shr_infnan_mod  , only : nan => shr_infnan_nan, assignment(=)
```

Reply:
changed spval to nan

Reply:
Perfect!